target/linux/generic/pending-5.15/921-sh-add-CONFIG_CMDLINE_OVERRIDE-option.patch

```diff
@@ -12,14 +28,14 @@
 +      devices. This setting enables using "chosen/cmdline-override" as the
 +      cmdline if it exists in the device tree.
 +
- config EXTRA_TARGETS
- 	string "Additional default image types"
- 	help
+ endmenu
+ 
+ menu "Bus options"
 --- a/drivers/of/fdt.c
 +++ b/drivers/of/fdt.c
-@@ -1162,6 +1162,17 @@ int __init early_init_dt_scan_chosen(uns
+@@ -1159,6 +1159,17 @@ int __init early_init_dt_scan_chosen(uns
  	if (p != NULL && l > 0)
- 		strlcat(data, p, min_t(int, strlen(data) + (int)l, COMMAND_LINE_SIZE));
```

Comment:
This is actually different... why `strlen(data) + i` ? we are not appending here so why we are accounting for strlen of data?

Reply:
l is also cast to int. strlen returns size_t...

Reply:
When this patch was in target/linux/{ipq806x,mediatek,mpc85xx}/patches-5.15, [target/linux/generic/hack-5.15/920-device_tree_cmdline.patch](https://github.com/openwrt/openwrt/blob/ce014b1e3903887d47bede8128f2c698418ac5f1/target/linux/generic/hack-5.15/920-device_tree_cmdline.patch) would have been applied before it, and this context line immediately before what this patch adds would have been an append, for `bootargs-append`.

With this moving to target/linux/generic/pending-5.15, it’s applied before hack-5.15, so the patch context changes.

Patch application order is controlled by [include/kernel-build.mk](https://github.com/openwrt/openwrt/blob/53a08e37437972ba0a8fbf953a93a70a6b784ef4/include/kernel-build.mk#L12).